corehq/apps/locations/models.py

```diff
@@ -833,6 +837,61 @@ def from_locations(cls, locations):
         related_locations = {loc_id for relation in relations for loc_id in relation}
         return related_locations - {l.location_id for l in locations}
 
+    @classmethod
+    def relation_distance_dictionary(cls, locations):
+        """Returns a dictionary of the related locations and their distance to each
+        location that's passed in if it has a relation.
+
+        If two locations that are passed in that have a relation. That relation is ignored.
+
+        {
+          related_location: {
+            loc_a_passed_in: distance,
+            loc_b_passed_in: distance
+          }
+        }
+        """
+        relations = cls.objects.filter(
+            Q(location_a__in=locations) | Q(location_b__in=locations)
+        ).prefetch_related('location_a', 'location_b')
```

Comment:
why is the `prefetch` needed? It doesn't look like any attributed from either of the fetched locations are used. Also this could be a `select_related` which would do a join instead of separate queries.

Reply:
Good question. Could probably be removed in that case

Reply:
https://github.com/dimagi/commcare-hq/pull/23311